src/ast/nodes/shared/FunctionBase.ts

```diff
@@ -115,6 +157,7 @@ export default abstract class FunctionBase extends NodeBase {
 					this.addArgumentToBeDeoptimized(argument);
 				}
 			}
+			this.trackArguments(args);
 			this.updateParameterVariableValues(args);
```

Comment:
`trackArguments` and `updateParameterVariableValues` are only called once here together, and their purpose is both to update parameter values in `ParameterVariable`. My feeling is that those two methods should be merged (e.g. to `updateParameterVariableValues`), and also that `ParameterVariable.updateKnownValue` and `ParameterVariable.trackArgument` should be merged. Or are there good reasons to keep them separate? Of course you now add the `pathKey`, but I do not think this is a problem.

Reply:
Has been merged into `updateParameterVariableValues `.